specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2023-07-01-preview/virtualMachineInstances.json

```diff
@@ -0,0 +1,1545 @@
+        },
+        "powerState": {
+          "type": "string",
+          "description": "The power state of the virtual machine instance",
+          "enum": [
+            "Deallocated",
+            "Deallocating",
+            "Running",
+            "Starting",
+            "Stopped",
+            "Stopping",
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "PowerStateEnum",
+            "modelAsString": true
+          }
+        },
+        "provisioningStatus": {
+          "type": "object",
+          "properties": {
+            "operationId": {
+              "type": "string",
+              "description": "The ID of the operation performed on the virtual machine instance"
+            },
+            "status": {
+              "type": "string",
+              "description": "The status of the operation performed on the virtual machine instance [Succeeded, Failed, InProgress]",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "InProgress"
+              ],
+              "x-ms-enum": {
+                "name": "Status",
+                "modelAsString": true
+              }
+            }
+          }
+        }
+      },
+      "description": "The observed state of virtual machine instances"
+    },
+    "VirtualMachineInstance": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "x-ms-azure-resource": true,
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineInstanceProperties"
+        },
+        "extendedLocation": {
+          "$ref": "./common.json#/definitions/ExtendedLocation",
+          "description": "The extendedLocation of the resource."
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Identity"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "The virtual machine instance resource definition."
+    },
+    "VirtualMachineInstanceView": {
+      "type": "object",
+      "properties": {
+        "vmAgent": {
+          "$ref": "#/definitions/VirtualMachineVMConfigAgentInstanceView",
+          "description": "The VM Config Agent running on the virtual machine."
+        }
+      },
+      "description": "The instance view of a virtual machine."
+    },
+    "VirtualMachineVMConfigAgentInstanceView": {
+      "type": "object",
+      "properties": {
+        "vmVMConfigAgentVersion": {
+          "type": "string",
+          "description": "The VM Config Agent full version."
+        },
+        "statuses": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InstanceViewStatus"
+          },
+          "x-ms-identifiers": [],
```

Comment:
>[] [](http://example.com/codeflow?start=30&length=2)

is it possible to use `code` as the identifying property?

Reply:
@pathuang-sia Is an `InstanceViewStatus`' `code` guaranteed to be unique in the array of `InstanceViewStatuses`?

Reply:
compute kept it empty so we did the same to algin with it: https://github.com/Azure/azure-rest-api-specs-pr/blob/a3089b2764df11b10a8a43e2a5a33dbe4d3653ca/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/virtualMachine.json#L1939

Is this a necessary change @ms-zhenhua? We understand it may be an improvement, but we would rather model over compute as how we did for previous changes suggested if this is not absolute necessary.

Reply:
currently it's optional. But if it is not set, the elements may be duplicated which may bring unexpected errors.